iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/service/ConfigNode.java

```diff
@@ -127,17 +126,17 @@ public void active() {
       if (SystemPropertiesUtils.isRestarted()) {
         LOGGER.info("{} is in restarting process...", ConfigNodeConstant.GLOBAL_NAME);
 
-        int configNodeId;
-        if (!SystemPropertiesUtils.isSeedConfigNode()) {
-          // The non-seed-ConfigNodes should send restart request and be checked (ip and port) by
-          // leader before initConsensusManager
-          sendRestartConfigNodeRequest();
-          configNodeId = CONF.getConfigNodeId();
-        } else {
-          configNodeId = SEED_CONFIG_NODE_ID;
-        }
+        int configNodeId = CONF.getConfigNodeId();
         configManager.initConsensusManager();
+        while (configManager.getConsensusManager().getLeader() == null) {
+          // leader has not been elected yet, wait 1 second
+          LOGGER.info("Leader has not been elected yet, wait for 1 second");
+          try {
+            Thread.sleep(1000);
+          } catch (Exception e) {
 
+          }
```

Comment:
log error and break this while?

Reply:
seems other similar 'exception during sleeping' will print warning log, so I do the same here.